### PR TITLE
#47 — Ability to edit user profile

### PR DESCRIPTION
**Pull request by sokoji**

### CODE REVIEW

File: models/User.php [hunk 1]
```diff
@@ -67,6 +76,26 @@ public function rules()
...
+            ['avatarFile', 'file', 'extensions' => 'png, jpg'],
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public function scenarios()
+    {
+        static $defaultAttributes = ['username', 'fullname', 'avatar', 'status'];
```
Comment (samdark) [inline models/User.php]: Status should not be in default attributes, only in manage.

File: migrations/m171028_191358_add_avatar_for_user.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+use yii\db\Migration;
+
+class m171028_191358_add_avatar_for_user extends Migration
+{
+    public function safeUp()
+    {
+        $this->addColumn('{{%user}}', 'avatar', $this->string(60)->after('fullname'));
```
Comment (samdark) [inline migrations/m171028_191358_add_avatar_for_user.php]: Why 60?

File: models/User.php (hunk 1 above)
Comment (samdark) [reply models/User.php]: Avatar isn't set directly so I think it's a bad idea to list it here.

File: views/layouts/main.php [hunk 3]
```diff
@@ -28,111 +28,121 @@
...
+                        </div>
+                    </div>
+                </div>
+
+                <?= $content ?>
             </div>
         </div>
-    </footer>
+
+        <footer class="footer">
```
Comment (samdark) [inline views/layouts/main.php]: Why class footer instead of tag footer?

File: migrations/m171028_191358_add_avatar_for_user.php (hunk 2 above)
Comment (sokoji) [reply migrations/m171028_191358_add_avatar_for_user.php]: update it to 16;
max int value for user_id plus file extension => '123123.jpg'

16 chars will be enough

File: models/User.php (hunk 1 above)
Comment (sokoji) [reply models/User.php]: fix it

File: views/layouts/main.php (hunk 3 above)
Comment (sokoji) [reply views/layouts/main.php]: remove class and fix less-file

File: migrations/m171028_191358_add_avatar_for_user.php [hunk 4]
```diff
@@ -6,7 +6,7 @@ class m171028_191358_add_avatar_for_user extends Migration
 {
     public function safeUp()
     {
-        $this->addColumn('{{%user}}', 'avatar', $this->string(60)->after('fullname'));
+        $this->addColumn('{{%user}}', 'avatar', $this->string(16)->after('fullname'));
```
Comment (samdark) [inline migrations/m171028_191358_add_avatar_for_user.php]: I was talking about increasing it, not decreasing :)

Comment (sokoji) [reply migrations/m171028_191358_add_avatar_for_user.php]: How much chars will be enough? default 255?

File: views/layouts/main.php [hunk 5]
```diff
@@ -28,111 +28,121 @@
...
-            $menuItems[] = ['label' => Yii::$app->user->identity->username, 'url' => ['/user/view', 'id' => \Yii::$app->user->id]];
-            $menuItems[] = [
-                'label' => Html::tag('span', '<span> ' . Yii::t('app', 'Logout') . '</span>', ['class' => 'fa fa-sign-out']),
-                'url' => ['/site/logout'],
-                'encode' => false,
-                'linkOptions' => [
-                    'data-method' => 'post',
-                    'alt' => Yii::t('app', 'Logout'),
-                    'title' => Yii::t('app', 'Logout'),
+        <div class="content">
```
Comment (samdark) [inline views/layouts/main.php]: What's this change for? What behavior does it fix?

Comment (sokoji) [reply views/layouts/main.php]: It's for footer-fix. Footer places at the bottom by "display: flex".
div.content tag need for flex elements.

File: migrations/m171028_191358_add_avatar_for_user.php (hunk 4 above)
Comment (samdark) [reply migrations/m171028_191358_add_avatar_for_user.php]: Yep. No need for less chars.

Comment (sokoji) [reply migrations/m171028_191358_add_avatar_for_user.php]: Fix this issue.

File: views/user/update.php [hunk 6]
```diff
@@ -6,15 +6,17 @@
 /* @var $model app\models\User */
 
 $this->title = Yii::t('user', 'Update {modelClass}: ', [
-    'modelClass' => 'User',
+    'modelClass' => 'пользователя',
```
Comment (samdark) [inline views/user/update.php]: Doesn't look right. It will be "пользователя" in English version.

Comment (samdark) [reply views/user/update.php]: I'd adjust whole string as

```php
$this->title = Yii::t('user', 'Update user {username}', ['username' => $model->username]);
```

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request closed without merging**